pkg/hintrunner/zero/zerohint.go

```diff
@@ -96,3 +160,36 @@ func GetParameters(zeroProgram *zero.ZeroProgram, hint zero.Hint, hintPC uint64)
 
 	return cellRefParams, resOpParams, nil
 }
+
+func applyApTracking(p *zero.ZeroProgram, h zero.Hint, ref zero.Reference, v any) any {
+	// We can't make an inplace modification because the v's underlying type is not a pointer type.
+	// Therefore, we need to return it from the function.
+	// This makes this function less elegant: it requires type asserts, etc.
```

Comment:
What would be a return type of Evaluate?

Reply:
It took me some time to understand the idea bug I get it now.
Does it make sense to implement it as a part of ID manager feature?
It might be connected due to the fact that we need to use these resolved values from the manager.

Reply:
Yes sure, I'll add this as part of the ID manager feature.